MetaMask/metamask-extension
# Request queue 2

## **Description**

Dapps wanting to access different networks can now rejoice. They will no longer need to detect changes to the currentnetwork and call switchEthereumChain. For methods with confirmations, this done for them automatically. For non-confiirmation requests, they will go straight through with no popups and no requirement to switch the network. Woohoo.

To use this feature, you must toggle the setting in the experimental settings tab.

## **Related issues**

Fixes: #

## **Manual testing steps**

1. Send a request (wallet watch asset for example) from dapp a
2. change the network from dapp b
3. send request (send transactino for example) from dapp b
4. approve the request from dapp a
5. note the automatic switching
6. send a non-confimation request (eth_chainId) from dapp a
7. note that it works without switching chain.
8. approve request for dapp b
9. ~win~ profit

## **Screenshots/Recordings**


https://youtu.be/2ZEAjDb6MGI

### **Before**

before & after both captured in the 1 video


### **After**

before & after both captured in the 1 video


## **Pre-merge author checklist**


## **Pre-merge reviewer checklist**
